### DISCUSSION TITLE
Flat Config: Ignores

### DISCUSSION THREAD

**Discussion by msalsbery** [eslint/eslint#17429] [Q&A] [answered]
I've been going insane trying to get this to work....seems like it should be obviously simple and common...

I've converted a couple projects to the new config. I run eslint CLI from a root folder (where eslint.config.js resides) something like `npx eslint **/*` or `npm run lint` (where the "lint" script is `eslint **/*`). 

There's a folder off the project root called `build`. I don't want any js file in that folder or any of its subfolders linted.

Does anyone have an example of the simplest eslint.config.js that ignores a folder called build?

**Comment by msalsbery** [comment]
Ok, obviously this works
```js
export default [
  js.configs.recommended,
  {
    ignores: ['build/'],
  },
];
```
Where it starts to fail is when I thought I was adding "global" language options
```js
export default [
  js.configs.recommended,
  {
    ignores: ['build/'],
    languageOptions: {
      ecmaVersion: 2020,
      sourceType: 'module', // module, script, commonjs
      globals: {
        ...globals.es2020,
        ...globals.node,
        ...globals.shelljs,
      },
    },
  },
];
```
ignores is ignored as soon as I add languageOptions there. Since my projects have Typescript as well always, I moved the language options to a `files: ['**/*.{js,mjs,cjs}']` config object and the ignores is now still working.

Clearly I don't have a firm grasp on how this flat config works yet...

**Comment by msalsbery** [comment, marked as answer]
Well shoot...

`If ignores is used without any other keys in the configuration object, then the patterns act as global ignores`

`Note that only global ignores patterns can match directories. ignores patterns that are specific to a configuration will only match file names`

RTFM for the 6th time and read ALL the sentences (and have a public discussion with yourself) to ensure success!

**Reply by elliot-huffman** [reply]
same, I read it through it just didn't click for me. Thank you for this post, it helped a ton!

**Comment by cmahnke** [comment]
I've been bitten by this to. It took me some time to find this post. The current way is quite suboptimal. It would be easier to decouple meaning and semantics here:
* There should be a `globalIgnores` which works regardless of it's suroundigs. That way it would be easier to understand (and override) what's going here. The current way is way less to expressive...

**Comment by ShortRoundDev** [comment]
Also having this issue, but with `files`. Setting `files: ["src/**/*"]` appears to do nothing now because of the idiosyncracies of the flat config structure. What is the correct way to tell eslint to only lint files in the src/ directory, for all configurations (including the preset ones)? i.e: the flat structure equivalent of the old configuration system? Here is my configuration, and to me it makes no sense that I would be seeing linting errors from anything other than my src/ directory:

```js
export default [
  pluginJs.configs.recommended,
  ...tseslint.configs.recommended,
  pluginReact.configs.flat.recommended,
  {
    files: ['src/**/*.{js,jsx,ts,tsx}'],
    languageOptions: { globals: { ...globals.browser, ...globals.node} },
  }
];
```

Part of this configuration was generate with eslint --init as well. The documentation says:

> Because config objects that don’t specify `files` or `ignores` apply to all files that have been matched by any other configuration object, they will apply to all JavaScript files

Which is indecipherable to me. It seems like the new configuration format didn't come up with a way to globally ignore or target a particular directory?

**Reply by cmahnke** [reply]
This is an answered question - I suppose only people with the problem come here via Google. I'm not sure if the developers see this. Maybe we should see if someone already has raised an issue on this.
And if not, we should open one with a good proposal.
